include/csoundCore.h

```diff
@@ -660,6 +662,199 @@ typedef int32_t (*SUBR)(CSOUND *, void *);
     INSDS   *insdshead;
   } OPDS;
 
+
+  static inline int32_t byte_order(void){
+    const int32_t one = 1;
+    return (!*((char*) &one));
+  }
+  
+  static inline int isstrcod(MYFLT xx){
+    int sel = (byte_order()+1)&1;
+#ifdef USE_DOUBLE
+    union {
+      double d;
+      int32_t i[2];
+    } z;
+    z.d = xx;
+    return ((z.i[sel]&0x7ff00000)==0x7ff00000);
+#else
+    union {
+      float f;
+      int32_t j;
+    } z;
+    z.f = xx;
+    return ((z.j&0x7f800000) == 0x7f800000);
+#endif
+  }
+  /** @name Opcode attributes */
+  /**@{ */
+
+  /**
+   * Returns true if argument is a string code
+   */
+  static inline int IsStringCode(MYFLT f){
+    return isstrcod(f);
+  }
+  
+  /**
+   * Returns the number of input arguments for opcode 'p'.
+   */
+  static inline int GetInputArgCnt(void *p){
```

Comment:
API-wise, seems clearer to me to make the argument OPDS* and make calling code case to OPDS* if necessary. (This applies to this function and others below.)

Reply:
ok sounds good to me